ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -491,19 +585,43 @@ private DeleteResult run_delete_v2(String accountName, String containerName, Str
     }
     // only need to issue an update statement if the row was not already marked as deleted.
     if (!alreadyDeleted) {
-      try (PreparedStatement statement = connection.prepareStatement(SOFT_DELETE_QUERY_V2)) {
-        // use the current time
-        statement.setTimestamp(1, currentTime);
-        statement.setInt(2, accountId);
-        statement.setInt(3, containerId);
-        statement.setString(4, blobName);
-        statement.setLong(5, version);
-        statement.executeUpdate();
-      }
+      applySoftDelete(accountId, containerId, blobName, version, currentTime, connection);
     }
     return new DeleteResult(blobId, alreadyDeleted);
   }
 
+  private List<StaleNamedResult> runPullStaleBlobIds(final Connection connection) throws Exception {
+    try (PreparedStatement statement = connection.prepareStatement(GET_STALE_QUERY)) {
+      try (ResultSet resultSet = statement.executeQuery()) {
+        List<StaleNamedResult> resultList = new ArrayList<>();
+        while (resultSet.next()) {
+          short accountId = resultSet.getShort(1);
+          short containerId = resultSet.getShort(2);
+          String blobName = resultSet.getString(3);
+          String blobId = Base64.encodeBase64URLSafeString(resultSet.getBytes(4));
+          long version = resultSet.getLong(5);
+          Timestamp currentTime = resultSet.getTimestamp(6);
+          StaleNamedResult result = new StaleNamedResult(accountId, containerId, blobName, blobId, version, currentTime);
+          resultList.add(result);
+        }
+        return resultList;
+      }
+    }
+  }
+
+  private void applySoftDelete(short accountId, short containerId, String blobName, long version, Timestamp currentTime,
+      Connection connection) throws Exception {
+    try (PreparedStatement statement = connection.prepareStatement(SOFT_DELETE_QUERY_V2)) {
+      // use the current time
+      statement.setTimestamp(1, currentTime);
```

Comment:
I would suggest to renaming currentTime to deleteTs. Since this is a common method for different use cases, it's better to name it in a more generic way to avoid confusion.

Reply:
Done, thanks!